Dockerfile

```diff
@@ -1,26 +1,22 @@
-FROM hashicorp/terraform:0.14.7
+ARG TERRAFORM_VERSION=0.14.7
+ARG RUBY_VERSION=3.1.2
 
-ENV BUILD_PACKAGES bash ruby-dev build-base
-ENV RUBY_PACKAGES ruby ruby-bundler
+FROM hashicorp/terraform:${TERRAFORM_VERSION} as terraform
 
-# Update and install all of the required packages.
-# At the end, remove the apk cache
-RUN apk --no-cache add $BUILD_PACKAGES && \
-    apk --no-cache add $RUBY_PACKAGES && \
-    gem install bigdecimal:3.0.2 && \
-    rm -rf /var/cache/apk/*
-
-RUN mkdir /usr/action
+FROM ruby:${RUBY_VERSION}
 WORKDIR /usr/action
-
-COPY Gemfile /usr/action/
-RUN bundle install
-
+RUN useradd -ms /bin/bash kitchen \
+  && chown kitchen:kitchen /usr/action
+RUN apt install ca-certificates
+USER kitchen
 COPY entrypoint.sh /entrypoint.sh
-
-# Need to run as root inorder to update CA certificates
+COPY --from=terraform /bin/terraform /bin/terraform
+COPY --chown=kitchen:kitchen Gemfile Gemfile.lock /usr/action/
```

Comment:
by copying these into `/usr/action` it leads to them being overwritten in the test (which mounts a volume over `/usr/action`) and I suspect it will do the same when used in other Actions. If they need to be in this location (I suspect they do), then maybe but them elsewhere during Docker build, then have the `entrypoint.sh` copy them from that location to `/usr/action`, which will happen after the volume is mounted

Reply:
This is now copied to `/usr/kitchen` but never used.